### PR TITLE
add possible images/templates to datacenter.yaml for Openstack and vSphere

### LINKED ISSUE #1297: Make vsphere templates configurable
Currently the vsphere template name is a free-text field, we should adjust this to be a configured list, e.G. via frontend config from the backend

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
This PR will add a static definition of operating system & image/template to the datacenters.yaml definition for vSphere and OpenStack.
The Dashboard should use this information to prefill the required inputs or offer a dropdown.

Fixes #1297 

**Release note**:
```release-note cloud provider
Image templates can now be configured in datacenter.yaml for Openstack and vSphere
```

### CODE REVIEW

**Review by alvaroaleman — COMMENTED** [review]
Wouldn't it be possible to use a map[OperatingSystem] string and hence skip the validation?

**Comment by alvaroaleman** [comment]
And maybe just import the operating system constants from the machine-controller?

**Comment by mrIncompetent** [comment]
> Wouldn't it be possible to use a map[OperatingSystem] string and hence skip the validation?

Can you do a comment on the code?

**Comment by mrIncompetent** [comment]
> And maybe just import the operating system constants from the machine-controller?

Yeah, that's probably a good idea

File: api/pkg/api/v1/types.go [hunk 1]
```diff
@@ -28,12 +28,16 @@ type HetznerDatacenterSpec struct {
 	Location   string `json:"location"`
 }
 
+// ImageList defines a map of operating system and the image to use
+type ImageList map[string]string
```
Comment (alvaroaleman) [inline api/pkg/api/v1/types.go:32]: Could we maybe define this as `map[OperatingSystem]string`?

Comment (mrIncompetent) [reply api/pkg/api/v1/types.go:32]: We could - but this is just a stupid api representation object. So a stripped down copy of the `DatacenterMeta`.
Defining here `map[OperatingSystem]string` would not do us any good. We would need to copy via a loop anyway

Comment (alvaroaleman) [reply api/pkg/api/v1/types.go:32]: The goal was primarily to elimante the need for the `validateImageList` func

Comment (mrIncompetent) [reply api/pkg/api/v1/types.go:32]: We would still need the `validateImageList` function as Unmarshalling does not care if the value is `coreos/ubuntu/centos`

**Comment by mrIncompetent** [comment]
@alvaroaleman PTAL

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**